api/src/test/java/org/pmiops/workbench/db/jdbc/ReportingNativeQueryServiceTest.java

```diff
@@ -0,0 +1,132 @@
+package org.pmiops.workbench.db.jdbc;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import javax.persistence.EntityManager;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.dao.CohortDao;
+import org.pmiops.workbench.db.dao.DataSetDao;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.dao.projection.ProjectedReportingUser;
+import org.pmiops.workbench.db.dao.rowmappers.ReportingDatasetCohortRowMapperImpl;
+import org.pmiops.workbench.db.model.DbCdrVersion;
+import org.pmiops.workbench.db.model.DbCohort;
+import org.pmiops.workbench.db.model.DbDataset;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.model.ReportingDatasetCohort;
+import org.pmiops.workbench.model.ReportingUser;
+import org.pmiops.workbench.testconfig.ReportingTestUtils;
+import org.pmiops.workbench.testconfig.fixtures.ReportingTestFixture;
+import org.pmiops.workbench.testconfig.fixtures.ReportingUserFixture;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Test the unique ReportingNativeQueryService, which bypasses Spring in favor of low-level JDBC
+ * queries. This means we need real DAOs.
+ */
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class ReportingNativeQueryServiceTest {
+
+  @Autowired private ReportingNativeQueryService reportingNativeQueryService;
+
+  // It's necessary to bring in several Dao classes, since we aim to populate join tables
+  // that have neither entities of their own nor stand-alone DAOs.
+  @Autowired private CdrVersionDao cCdrVersionDao;
```

Comment:
We could almost pull all of these machinations into a reporting config class, since we'll need to set up similar things for other joint tables. To start with, I think it's fine to overload this test class with dependencies, as join tables always require extra test data to test.